tests/DCPS/TimeBasedFilter/main.cpp

```diff
@@ -576,6 +586,19 @@ ACE_TMAIN(int argc, ACE_TCHAR** argv)
 
     SampleMap send_map;
 
+    // receiving the first safety profile sample takes longer than
+    // subsequent samples, so write a throwaway
```

Comment:
Just for my own understanding, why does the first sample take longer than the others?

Reply:
I checked with Adam and think it is due to RTPS discovery which is only used in safety profile.